test.py

```
@@ -0,0 +1,716 @@
+            for p in [25, 50, 75, 90]:
+                stats[f'p{p}'] = {
+                    'initial_contrast': self._percentile(initial_contrasts, p),
+                    'delta_e': self._percentile(delta_es, p),
+                }
+            return stats
+
+        overall_success_stats = get_percentile_stats(actual_success)
+        overall_failure_stats = get_percentile_stats(actual_failure)
+
+        # --- 3. Initial State Categorization (NEW ANALYSIS) ---
+        initial_state_analysis: Dict[str, Dict[str, List[Dict]]] = {
```

Ruff (0.14.6)

Local variable `initial_state_analysis` is assigned to but never used

Remove assignment to unused variable `initial_state_analysis`

(F841)